src/main/java/net/minecraftforge/client/gui/overlay/ForgeLayeredDraw.java

```diff
@@ -0,0 +1,337 @@
+    public static final ResourceLocation     BOSS_OVERLAY = ResourceLocation.withDefaultNamespace("boss_overlay");
+
+    public static final ResourceLocation POST_SLEEP_PHASE = ResourceLocation.withDefaultNamespace("post_sleep_phase");
+    public static final ResourceLocation     DEMO_OVERLAY = ResourceLocation.withDefaultNamespace("demo");
+    public static final ResourceLocation    DEBUG_OVERLAY = ResourceLocation.withDefaultNamespace("debug");
+    public static final ResourceLocation       SCOREBOARD = ResourceLocation.withDefaultNamespace("scoreboard");
+    public static final ResourceLocation   HOTBAR_MESSAGE = ResourceLocation.withDefaultNamespace("hotbar_message");
+    public static final ResourceLocation    TITLE_OVERLAY = ResourceLocation.withDefaultNamespace("title");
+    public static final ResourceLocation     CHAT_OVERLAY = ResourceLocation.withDefaultNamespace("chat_overlay");
+    public static final ResourceLocation         TAB_LIST = ResourceLocation.withDefaultNamespace("tab_list");
+    public static final ResourceLocation SUBTITLE_OVERLAY = ResourceLocation.withDefaultNamespace("subtitle");
+
+    public static final ResourceLocation    COMBINE_PHASE = ResourceLocation.withDefaultNamespace("combine_phase");
+    public static final ResourceLocation    SLEEP_OVERLAY = ResourceLocation.withDefaultNamespace("sleep_overlay");
+
+
+    public static final ImmutableList<ResourceLocation> PRE_LIST = ImmutableList.of(CAMERA_OVERLAY, CROSSHAIR, HOTBAR, EXPERIENCE, POTION_EFFECTS, BOSS_OVERLAY);
+    public static final ImmutableList<ResourceLocation> POST_LIST = ImmutableList.of(DEMO_OVERLAY, DEBUG_OVERLAY, SCOREBOARD, HOTBAR_MESSAGE, TITLE_OVERLAY, CHAT_OVERLAY, TAB_LIST, SUBTITLE_OVERLAY);
+    public static final ImmutableList<ResourceLocation> COMBINE_LIST = ImmutableList.of(PRE_SLEEP_PHASE, SLEEP_OVERLAY, POST_SLEEP_PHASE);
+
+    /**
+     * Creates a stack of pre-named layers for vanilla.
+     * Not intended for modder use. Use {@linkplain ForgeLayeredDraw(ResourceLocation, Boolean, String...)} instead.
+     * @param phase Phase indicator
+     * @param layers locations of layers to add
+     */
+    @ApiStatus.Internal
+    public ForgeLayeredDraw(ResourceLocation phase, List<ResourceLocation> layers) {
+        this.phase = phase;
+        expectedNames.addAll(layers);
+    }
+
+    /**
+     * Creates an empty draw list. Add entries with {@linkplain ForgeLayeredDraw#add(ResourceLocation, Layer)}
+     * @param phase marker for which phase this is.
+     */
+    public ForgeLayeredDraw(ResourceLocation phase) {
+        this.phase = phase;
+    }
+
+    /**
+     * Adds a full, pre-named draw stack.
+     * Modders should be using {@linkplain ForgeLayeredDraw#add(ResourceLocation, ForgeLayeredDraw, BooleanSupplier)}
+     * @param layeredDraw layer stack to be added.
+     * @param booleanSupplier requirement for it to render.
+     * @return this
+     */
+    @ApiStatus.Internal
```

Comment:
You cant mark vanilla methods as internal. You can deprecate them but they cant be internal